src/server.py

```diff
@@ -26,26 +26,37 @@ def generate_ssl_cert(cert_path, key_path, config_path):
 context = ssl.SSLContext(ssl.PROTOCOL_TLS_SERVER)
 context.load_cert_chain(certfile=cert_path, keyfile=key_path)
 
-with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
-    s.bind(('localhost', 1234))
-    s.listen(5)
-    print("Server is listening on port 1234...")
+clients = []
+client_counter = 1
+
+async def handle_client(reader, writer):
+    global client_counter
+    client_id = client_counter
+    client_counter += 1
+    clients.append((reader, writer))
+    addr = writer.get_extra_info('peername')
+    print(f"Connection from {addr} has been established as client{client_id}!")
     
-    with context.wrap_socket(s, server_side=True) as ssock:
-        conn, addr = ssock.accept()
-        print(f"Connection from {addr} has been established!")
-        with conn:
-            try:
-                while True:
-                    data = conn.recv(1024)
-                    if not data:
-                        print("Connection closed by client")
-                        break
-                    print(f'client: {data.decode()}')
-                    resp = input('server: ')
-                    conn.sendall(resp.encode())
-
-            except ConnectionError as e:
-                print(f"Connection error occurred: {e}")
-
-            print("Server is shut down")
+    try:
+        while True:
```

Comment:
this loop was breaking after recieving the first msg from the client because it thinks the data is empty after recieving so i edited your code and guess what ?